app/code/community/TIG/MyParcel2014/Model/Api/MyParcel.php

```diff
@@ -856,26 +864,43 @@ public function getTotalWeight($totalWeight, $item, $isWoldShipment = false) {
         }
 
         $weight *= $qty;
-        if ($isWoldShipment != false){
-            $weight = max(array(1, $weight));
-        }
 
-        $totalWeight += $weight * 1000;
+        $totalWeight += $this->getCalculatedWeightToGram($weight, $myParcelShipment);
+
         $price *= $qty;
+        return ['weight' => $weight, 'total_weight' => $totalWeight, 'qty' => $qty, 'price' => $price];
+    }
 
+    /**
+     * @param                                  $weight
+     * @param \TIG_MyParcel2014_Model_Shipment $myParcelShipment
+     *
+     * @return int
+     */
+    protected function getCalculatedWeightToGram($weight, $myParcelShipment)
+    {
+        // @var TIG_MyParcel2014_Helper_Data $helper
```

Comment:
In PhpStorm werkt // niet met tags toch?
```suggestion
        /**
        *  @var TIG_MyParcel2014_Helper_Data $helper
        */
```
Als dat zo is moet je (als je tags gebruikt) in dit geval /** gebruiken

Reply:
Done